### ISSUE TITLE
[Linux] Cannot launch Dota 2 on arch linux ( stuck at processing vulkan shaders )

### ISSUE THREAD

**Issue by AntouanK** [ValveSoftware/Dota-2#2644]
Can't run Dota 2 at all on my system.
I tried everything I found on other issues, but nothing worked so far.
I used different flags, different proton versions, etc.

It either launches and crashes after the valve logo, or just gets stuck at validating or Processing the vulkan shaders ( stuck at 0% for a long time for example ).

very strange...

runtime diagnostics:
https://gist.github.com/AntouanK/668f6f7980dbaef877e7cc1725179d22

I have an nvidia 4090.

<details><summary>system info:</summary>

```
Computer Information:
Manufacturer: ASUSTeK COMPUTER INC.
Model: PRIME X670E-PRO WIFI
Form Factor: Desktop
No Touch Input Detected
Processor Information:
CPU Vendor: AuthenticAMD
CPU Brand: AMD Ryzen 9 7900 12-Core Processor
CPU Family: 0x19
CPU Model: 0x61
CPU Stepping: 0x2
CPU Type: 0x0
Speed: 5482 MHz
24 logical processors
12 physical processors
Hyper-threading: Supported
FCMOV: Supported
SSE2: Supported
SSE3: Supported
SSSE3: Supported
SSE4a: Supported
SSE41: Supported
SSE42: Supported
AES: Supported
AVX: Supported
AVX2: Supported
AVX512F: Supported
AVX512PF: Unsupported
AVX512ER: Unsupported
AVX512CD: Supported
AVX512VNNI: Supported
SHA: Supported
CMPXCHG16B: Supported
LAHF/SAHF: Supported
PrefetchW: Unsupported
Operating System Version:
"Manjaro Linux" (64 bit)
Kernel Name: Linux
Kernel Version: 6.6.8-2-MANJARO
X Server Vendor: The X.Org Foundation
X Server Release: 12101010
X Window Manager: i3
Steam Runtime Version: steam-runtime_0.20231024.64411
Video Card:
Driver: Mesa llvmpipe (LLVM 16.0.6, 256 bits)
Driver Version: 4.5 (Compatibility Profile) Mesa 23.1.9-manjaro1.1
OpenGL Version: 4.5
Desktop Color Depth: 24 bits per pixel
Monitor Refresh Rate: 71 Hz
VendorID: 0x10de
DeviceID: 0x2684
Revision Not Detected
Number of Monitors: 1
Number of Logical Video Cards: 2
Primary Display Resolution: 5120 x 2160
Desktop Resolution: 5120 x 2160
Primary Display Size: 36.61" x 15.35" (39.69" diag), 93.0cm x 39.0cm (100.8cm diag)
Primary Bus: PCI Express 16x
Primary VRAM: 11008 MB
Supported MSAA Modes: 2x 4x 8x 16x
Sound card:
Audio device: Nvidia GPU a3 HDMI/DP
Memory:
RAM: 63439 Mb
VR Hardware:
VR Headset: None detected
Miscellaneous:
UI Language: English
LANG: en_GB.UTF-8
Total Hard Disk Space Available: 283376 MB
Largest Free Hard Disk Block: 148472 MB
Storage:
Number of SSDs: 3
SSD sizes: 2000G,250G,250G
Number of HDDs: 0
Number of removable drives: 0
```

</details>

**Comment by kisak-valve** [comment]
Hello @AntouanK, please copy the contents of Steam Runtime Diagnostics from Steam (`Steam` -> `Help` -> `Steam Runtime Diagnostics`) and put it in a [gist](https://gist.github.com/), then include a link to the gist in this issue report.

It may be unrelated to the issue you're seeing with the game, but `Driver: Mesa llvmpipe (LLVM 16.0.6, 256 bits)` tells us that you're missing the 32 bit userspace libraries for your NVIDIA driver install and Steam is being forced to fallback to llvmpipe (mesa's faster CPU renderer) to run at all. Please add the `lib32-nvidia-utils` system package.

**Comment by AntouanK** [comment]
you are right, I was missing that library.
I installed `lib32-nvidia-utils-545.29.06-1`
Still, gets stuck at vulkan shaders

![image](https://github.com/ValveSoftware/Dota-2/assets/4569111/9e1cd6c7-847f-4c9a-b90a-131791baeb4e)

**Comment by AntouanK** [comment]
runtime diagnostics after I installed that library:
https://gist.github.com/AntouanK/ee8203146c2ddd2a08227511aa010218

<details><summary>system info:</summary>

```
Computer Information:
Manufacturer: ASUSTeK COMPUTER INC.
Model: PRIME X670E-PRO WIFI
Form Factor: Desktop
No Touch Input Detected
Processor Information:
CPU Vendor: AuthenticAMD
CPU Brand: AMD Ryzen 9 7900 12-Core Processor
CPU Family: 0x19
CPU Model: 0x61
CPU Stepping: 0x2
CPU Type: 0x0
Speed: 5482 MHz
24 logical processors
12 physical processors
Hyper-threading: Supported
FCMOV: Supported
SSE2: Supported
SSE3: Supported
SSSE3: Supported
SSE4a: Supported
SSE41: Supported
SSE42: Supported
AES: Supported
AVX: Supported
AVX2: Supported
AVX512F: Supported
AVX512PF: Unsupported
AVX512ER: Unsupported
AVX512CD: Supported
AVX512VNNI: Supported
SHA: Supported
CMPXCHG16B: Supported
LAHF/SAHF: Supported
PrefetchW: Unsupported
Operating System Version:
"Manjaro Linux" (64 bit)
Kernel Name: Linux
Kernel Version: 6.6.8-2-MANJARO
X Server Vendor: The X.Org Foundation
X Server Release: 12101010
X Window Manager: i3
Steam Runtime Version: steam-runtime_0.20231024.64411
Video Card:
Driver: NVIDIA Corporation NVIDIA GeForce RTX 4090/PCIe/SSE2
Driver Version: 4.6.0 NVIDIA 545.29.06
OpenGL Version: 4.6
Desktop Color Depth: 24 bits per pixel
Monitor Refresh Rate: 71 Hz
VendorID: 0x10de
DeviceID: 0x2684
Revision Not Detected
Number of Monitors: 1
Number of Logical Video Cards: 2
Primary Display Resolution: 5120 x 2160
Desktop Resolution: 5120 x 2160
Primary Display Size: 36.61" x 15.35" (39.69" diag), 93.0cm x 39.0cm (100.8cm diag)
Primary Bus: PCI Express 16x
Primary VRAM: 24564 MB
Supported MSAA Modes: 2x 4x 8x 16x
Sound card:
Audio device: Nvidia GPU a3 HDMI/DP
Memory:
RAM: 63439 Mb
VR Hardware:
VR Headset: None detected
Miscellaneous:
UI Language: English
LANG: en_GB.UTF-8
Total Hard Disk Space Available: 283376 MB
Largest Free Hard Disk Block: 148453 MB
Storage:
Number of SSDs: 3
SSD sizes: 2000G,250G,250G
Number of HDDs: 0
Number of removable drives: 0
```

**Comment by ArtyomZabroda** [comment]
I have the same problem on arch linux, yesterday it was working fine. Reinstalling doesn't help.

Runtime diagnostics:
https://gist.github.com/ArtyomZabroda/ead81053ef0411b7a826bee707409e57

**Comment by ArtyomZabroda** [comment]
-vulkan_disable_pipeline_cache   launch option seems to fix this issue

**Comment by AntouanK** [comment]
I'm still stuck at 0%.
do you skip that part?
or does it finish the processing after adding the flag in properties?

**Comment by AntouanK** [comment]
https://github.com/ValveSoftware/Dota-2/assets/4569111/7cffbf03-8d73-4109-8419-0104600c6898

**Comment by AntouanK** [comment]
Tried to uninstall the game and install it again.
Seems like now it's stuck on "validating" and it doesn't download it.
I cannot close the client as well or cancel the download.
I removed steam and installed it again, it starts "downloading" dota 2 right away and gets stuck at 0%.

Is this a steam issue or dota2 issue? I'm very confused.

**Comment by ArtyomZabroda** [comment]
Seems that I don't experience the same issue as you do. I found a solution that is working for me in this thread https://github.com/ValveSoftware/Dota-2/issues/2626 .

**Comment by ArtyomZabroda** [comment]
> I'm still stuck at 0%. do you skip that part? or does it finish the processing after adding the flag in properties?

After adding -vulkan_disable_pipeline_cache, game was working as expected with and without skipping

**Comment by AntouanK** [comment]
yes, might be the disk for me. I had all my steam library there.
I deleted everything and used another disk now and it's downloading the game for now.
so that might have been the issue(?)

I'll check if the game does play and come back to update.

**Comment by Alex-Beh** [comment]
check whether your GPU driver is running properly or not. Try run the following command if you using NVIDIA GPU
`
nvidia-smi
`

**Comment by t4fita** [comment]
> -vulkan_disable_pipeline_cache launch option seems to fix this issue

Thanks a lot this worked for me. 

As a reference, my issue was that I wasn't able to launch dota at all. It worked fine and I just took a 15 minutes break, closed dota, without updating anything or doing anything or even shutting down my computer or steam. After I came back, when clicking play, it says running for a few seconds and then immediately stops without even the screen popping in, no vulkan shaders processing or anything.
